android/app/src/androidTest/java/org/fossasia/openevent/DateTest.java

```diff
@@ -0,0 +1,122 @@
+package org.fossasia.openevent;
+
+import org.fossasia.openevent.utils.ISO8601Date;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Created by arpitdec5 on 02-06-2017.
+ */
+
+public class DateTest {
+
+    @Test
+    public void testGetTimeZoneDateStringFromString() {
+        ISO8601Date.setTimeZone("UTC");
+        ISO8601Date.setEventTimeZone("Asia/Singapore");
+
+        //date strings to be used for testing
+        String dateString1 = "2017-06-02T07:59:10Z";
+        String dateString2 = "2017-06-02T20:00:10Z";
+        String dateString3 = "2017-06-03T00:00:10Z";
+        String dateString4 = "2017-06-03T07:00:10Z";
+        String predictedString;
+        String actualString;
+
+        predictedString = ISO8601Date.getTimeZoneDateStringFromString(dateString1);
+        actualString = "Thu, 01 Jun 2017, 23:59, UTC";
+        Assert.assertEquals(actualString, predictedString);
+
+        predictedString = ISO8601Date.getTimeZoneDateStringFromString(dateString2);
+        actualString = "Fri, 02 Jun 2017, 12:00, UTC";
+        Assert.assertEquals(actualString, predictedString);
+
+        predictedString = ISO8601Date.getTimeZoneDateStringFromString(dateString3);
+        actualString = "Fri, 02 Jun 2017, 16:00, UTC";
+        Assert.assertEquals(actualString, predictedString);
+
+        predictedString = ISO8601Date.getTimeZoneDateStringFromString(dateString4);
+        actualString = "Fri, 02 Jun 2017, 23:00, UTC";
+        Assert.assertEquals(actualString, predictedString);
+    }
+
+    @Test
+    public void testGetTimeZoneDateStringForDayFragment() {
+        ISO8601Date.setTimeZone("UTC");
+        ISO8601Date.setEventTimeZone("UTC");
+
+        //date strings to be used for testing
+        String day1 = "2017-06-01";
+        String day2 = "2017-06-02";
+        String predictedString;
+        String actualString;
+
+        predictedString = ISO8601Date.getTimeZoneDateStringFromStringForDayFragment(day1);
+        actualString = "1 Jun";
+        Assert.assertEquals(actualString, predictedString);
+
+        predictedString = ISO8601Date.getTimeZoneDateStringFromStringForDayFragment(day2);
+        actualString = "2 Jun";
```

Comment:
This is actually expected and the above is actual

Please change others accordingly. Use `expected` instead of predicted to avoid confusion.

Also, make your test names show what they are expected to do. Like `shouldConvertLocalTimeZoneStringTo...`

In this way, you can break this tests into much smaller components. See the link I gave as a reference. A signle method can have multiple tests. It is easier to see what use case failed when debugging failed errors